crates/provider/src/builder.rs

```diff
@@ -437,11 +437,12 @@ impl<L, F> ProviderBuilder<L, F, Ethereum> {
         let anvil_layer = crate::layers::AnvilLayer::from(f(Default::default()));
         let url = anvil_layer.endpoint_url();
 
-        let wallet = alloy_signer_wallet::Wallet::from(anvil_layer.instance().keys()[0].clone());
+        let signer =
+            alloy_signer_local::LocalSigner::from(anvil_layer.instance().keys()[0].clone());
```

Comment:
drive-by, while we're here, should we populate this with more than 1 key?

Reply:
Thought this way trivial to add but noticed some unexpected behavior when implementing this due to the way BTreeMap sorts the addresses, therefore if you get an iterator `signer_addresses` the first address is not the first that was inserted. This may have unexpected consequences when working with derivations / mnemonics where insertion order matters to users. The default address, being assigned upon first insertion stays the same so no issue there.

I think it is out of scope for this ticket, happy to implement it in another.

Reply:
Fixed for now as suggested here: https://github.com/alloy-rs/alloy/pull/883#issuecomment-2165759893

Still something we may want to evaluate the behavior of in a future ticket